src/dataset.py

```diff
@@ -0,0 +1,68 @@
+from pathlib import Path
+
+import lightning as L
+import rasterio as rio
+import torch
+from torch.utils.data import DataLoader, Dataset
+
+
+class ClayDataset(Dataset):
+    def __init__(self, data_dir: Path, transform=None):
+        super().__init__()
+        self.data_dir = data_dir
+        self.chips_path = list(data_dir.glob("**/*.tif"))
+        self.transform = transform
+
+    def read_chip(self, chip_path):
+        ts = chip_path.parent.name
+        year, month, day = map(int, ts.split("-"))
+
+        chip = rio.open(chip_path)
+        bounds = chip.bounds
+        centroid_x = (bounds.left + bounds.right) / 2
+        centroid_y = (bounds.top + bounds.bottom) / 2
```

Comment:
Coordinates from `chip.bounds` are UTM coordinates and not latlon. You would need to reproject first.

Reply:
I made an implementation to compute lat/lon centroids from bounds in the first tiler version, have a look
https://github.com/Clay-foundation/model/blob/9a41e3d356ac69d13ad6a7c06b20d1d21d4c9898/tile_retriever.py#L97